ra/registration-authority.go

```diff
@@ -602,6 +608,12 @@ func (ra *RegistrationAuthorityImpl) UpdateRegistration(base core.Registration,
 
 // UpdateAuthorization updates an authorization with new values.
 func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization, challengeIndex int, response core.Challenge) (authz core.Authorization, err error) {
+	// Refuse to update expired authorizations
+	if base.Expires == nil || base.Expires.Before(ra.clk.Now()) {
+		err = fmt.Errorf("Expired authorization")
```

Comment:
I'm fine returning StatusNotFound (404).

I don't really think there's a race condition, since the RA check is on data that the WFE hands it, not data from the SA.  So the only way it can happen is if the WFE screws up.  I'll compromise, though, by having the RA return NotFoundError, so it will look the same to the client.

I don't think we should the checks into the RA.  We should fail as fast as possible here.  Ideally, we could push it down to the SA, so it would just never return an expired authz.  But that's not critical and it doesn't look trivial, so I'm punting on that for now.